### PR TITLE
[OSJC-256] DefaultClient#equal <true> if authstrategies match in username

### PR DESCRIPTION
**Pull request by adietish**

### CODE REVIEW

**Comment by adietish** [comment]
[test]

**Comment by adietish** [comment]
@jcantrill dare to review?

**Comment by adietish** [comment]
[test]

**Comment by adietish** [comment]
weird, travis fails to build complainig that it cannot find the branch OSJC-256...

File: src/test/java/com/openshift/internal/restclient/DefaultClientTest.java [hunk 1]
```diff
@@ -165,18 +165,6 @@ public void clientShouldNotEqualClientWithDifferentUrl() throws Exception {
...
-		tokenClient.setAuthorizationStrategy(new TokenAuthorizationStrategy("aToken", "aUser"));
-
-		DefaultClient basicAuthClient = new DefaultClient(baseUrl, null);
-		basicAuthClient.setAuthorizationStrategy(new BasicAuthorizationStrategy("aUser", "aPassword", "aPassword"));
-
-		assertThat(tokenClient).isNotEqualTo(basicAuthClient);
-	}
-
-	@Test
 	public void client_should_equal_client_with_same_TokenAuthStrategy_with_different_token() throws Exception {
```
Comment (jcantrill) [inline src/test/java/com/openshift/internal/restclient/DefaultClientTest.java]: @adietish Looks like you lost a test

**Comment by jcantrill** [comment]
LGTM.  Per our conversation, I think equality should be obtained when the username and server match.   Other then verifying the 'lost' test I think this is ok to merge.

File: src/test/java/com/openshift/internal/restclient/DefaultClientTest.java (hunk 1 above)
Comment (adietish) [inline src/test/java/com/openshift/internal/restclient/DefaultClientTest.java]: @jcantrill nope, that's on purpose. See the method name, it says that 2 clients with different strategies should not be equal, that's exactly the opposite of what I wanted to change, so the test has to disappear.

**Comment by adietish** [comment]
[merge]

**Pull request merged**